### PR TITLE
auditbeat 6.1.1 (new formula)

### PR DESCRIPTION
**Pull request by jasontedor**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jasontedor** [comment]
```
18:18:30 Error: 2 problems in 1 formula
18:18:30 auditbeat:
18:18:30   * New formulae should not require patches to build. Patches should be submitted and accepted upstream first.
18:18:30   * New formulae should not require patches to build. Patches should be submitted and accepted upstream first.
```

The default configuration file that upstream ships with enables features that are not compatible with macOS so these need to be disabled, hence the patch here. Can we please whitelist this being patched?

**Comment by ilovezfs** [comment]
It would be far preferable to get a compatible config file into the tarball :)

**Comment by jasontedor** [comment]
I can look into that but that needs to be a follow-up, for now this is what we have for 6.0.0. 😇

**Comment by ilovezfs** [comment]
you'll need to use `inreplace` instead of a patch

**Comment by jasontedor** [comment]
@ilovezfs Why? This is much easier to write and a lot clearer.

**Comment by ilovezfs** [comment]
https://github.com/Homebrew/brew/blob/d450b2fec793232226ae09db825a2ad8dab1becc/docs/Formula-Cookbook.md#inreplace

>inreplace should be used instead of patches when patching something that will never be accepted upstream

Also, new formulae cannot have patches.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
This should use a resource block and `setup_install_args` instead of the virtualenv DSL.

**Comment by ilovezfs** [comment]
```
# Sets up the virtual python environment
.PHONY: python-env
python-env:
        @test -d $(PYTHON_ENV) || virtualenv $(VIRTUALENV_PARAMS) $(PYTHON_ENV)
        @$(PYTHON_ENV)/bin/pip install -q --upgrade pip autopep8 six
```
That will fetch binary wheels.

**Comment by ilovezfs** [comment]
`pip install --no-binary :all:` will prevent that.

**Comment by ilovezfs** [comment]
what's the status here?

**Comment by jasontedor** [comment]
@ilovezfs I have updated this PR; when you have a chance would you please review again?

File: Formula/auditbeat.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      (etc/"auditbeat").install Dir["auditbeat*.yml"]
+      prefix.install_metafiles
+    end
+
+    (bin/"auditbeat").write <<~EOS
+      #!/bin/sh
+        exec #{libexec}/bin/auditbeat \
+        -path.config #{etc}/auditbeat \
+        -path.data #{var}/lib/auditbeat \
+        -path.home #{prefix}/libexec \
```
Comment (ilovezfs) [inline Formula/auditbeat.rb]: #{libexec}

File: Formula/auditbeat.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
+class Auditbeat < Formula
+  desc "Lightweight Shipper for Audit Data"
+  homepage "https://www.elastic.co/products/beats/auditbeat"
+  url "https://github.com/elastic/beats/archive/v6.1.1.tar.gz"
+  sha256 "c69f0047644be919e42a1d8fa3383c894ca8e054d5b6f727f161ed4ce497ca84"
+
```
Comment (ilovezfs) [inline Formula/auditbeat.rb]: no blank line

File: Formula/auditbeat.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    EOS
+    pid = fork do
+      exec "#{bin}/auditbeat", "-path.config", testpath/"config", "-path.data", testpath/"data"
+    end
+    sleep 5
+
+    begin
+      touch testpath/"files/touch"
+      sleep 30
+      s = (testpath/"auditbeat/auditbeat").read.chomp
```
Comment (ilovezfs) [inline Formula/auditbeat.rb]: why chomp if we're not using `assert_equal`?

**Review by ilovezfs — APPROVED** [review]
minor stuff but :+1:

**Comment by jasontedor** [comment]
Thanks. I pushed a commit addressing your feedback.

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @jasontedor! :ship:d

**Comment by jasontedor** [comment]
Thanks @ilovezfs!

**Pull request closed without merging**